micmac_li3ds/api.py

```diff
@@ -7,21 +7,47 @@
 class Api(object):
 
     def __init__(self, api_url, api_key):
-        if not api_url:
-            err = 'Error: no api url provided'
-            raise ValueError(err)
-
-        if not api_key:
-            err = 'Error: no api key provided'
-            raise ValueError(err)
-
-        self.api_url = api_url.rstrip('/')
-        self.headers = {
-            'Accept': 'application/json',
-            'X-API-KEY': api_key
+        self.api_url = None
+        self.headers = None
+        self.staging = None
+        self.unique_keys = {
+            'transfo': ['name', 'source', 'target'],
+            'transfos/type': ['name'],
+            'transfotree': ['name', 'transfos'],
+            'referential': ['name', 'sensor'],
+            'sensor': ['name'],
+        }
+
+        if api_url and api_key:
+            self.api_url = api_url.rstrip('/')
+            self.headers = {
+                'Accept': 'application/json',
+                'X-API-KEY': api_key
+                }
+
+        else:
+            if api_key:
+                err = 'Error: no api url provided'
```

Comment:
The error message is not consistent with the test.

What don't you just use `api_url` to determine whether staging is activated or not? Something like this:

```py
    if api_url:
        if not api_key:
             err = 'Error: no api key provided'
             raise ValueError(err)
        self.api_url = api_url.rstrip('/')
        self.headers = {
            'Accept': 'application/json',
            'X-API-KEY': api_key
        }
    else:
        self.staging = {
            …
        }
```